Restrict cipher suites for Desktop Access in FIPS mode (#42277)
gravitational/teleport: docs/pages/access-controls/compliance-frameworks/fedramp.mdx

- All uses of non-compliant algorithms such as NaCl are removed and replaced with compliant algorithms such as AES-GCM.
- Teleport is compiled with [BoringCrypto](https://csrc.nist.gov/projects/cryptographic-module-validation-program/certificate/4407).
- User, host, and CA certificates (and host keys for recording proxy mode) only use 2048-bit RSA private keys.

> ## Desktop Access

Reviewer: ```suggestion
## Remote desktop access
```

> Teleport's Desktop Access feature uses Rust for RDP connections, and thus uses a fork of Cloudflare's [`boring`](https://github.com/gravitational/boring)

Reviewer: ```suggestion
Teleport uses Rust for RDP connections, and thus uses a fork of Cloudflare's [`boring`](https://github.com/gravitational/boring)
```

library under the hood for FIPS-compliant TLS cryptography. The primary noteable difference to the specifications listed above is that
TLS is restricted to TLS 1.2 only (1.3 is not supported).

> Note that Teleport's Desktop Access feature is not available in FIPS mode on `arm64`.

Reviewer: ```suggestion
Note that `arm64` FIPS builds do not support access to Windows desktops.
```
